### PR TITLE
PRO-1549: SDK E2E Pre-test Balance Check and modified the test cases

### PR DESCRIPTION
**Pull request by Jineshdarjee**
## Description

1. Implement the SDK E2E Pre-test Balance Check task.
2. Modified the test cases.

### CODE REVIEW

File: test/specs/mainnet/01_sendToken/Arbitrum.spec.js [hunk 1]
```diff
@@ -45,637 +50,556 @@ describe("The SDK, when sending a native token with arbitrum network on the Main
...
-      console.log(e);
-      assert.fail("The estimation of the batch is not performed successfully.");
-    }
+      try {
+        assert.isNotEmpty(
+          AddTransactionToBatchOutput.requests[0].to,
+          "The To Address value is empty in the Batch Response."
+        );
+      } catch (e) {
+        console.log(e);
```
Comment (IAmKio) [inline test/specs/mainnet/01_sendToken/Arbitrum.spec.js:112]: Remember to use `console.error` when you are reporting an error

Comment (IAmKio) [reply test/specs/mainnet/01_sendToken/Arbitrum.spec.js:112]: Please change this throughout the code

**Review by IAmKio — CHANGES_REQUESTED** [review]
I'm assuming that all file changes are similar - i've left a few small changes but other than that it looks fine. Thanks

File: test/specs/mainnet/01_sendToken/Arbitrum.spec.js (hunk 1 above)
Comment (Jineshdarjee) [reply test/specs/mainnet/01_sendToken/Arbitrum.spec.js:112]: updated the code in all the test cases as per the comment.

File: test/specs/mainnet/01_sendToken/Arbitrum.spec.js [hunk 2]
```diff
@@ -45,637 +50,556 @@ describe("The SDK, when sending a native token with arbitrum network on the Main
...
+        assert.fail(
+          "The expected validation is not displayed when entered the invalid To Address while performing batch execution."
+        );
       }
-    } catch (e) {
-      console.log(e);
-      assert.fail(
-        "The expected validation is not displayed when entered the invalid To Address while performing batch execution."
+    } else {
+      console.log(
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet/01_sendToken/Arbitrum.spec.js:431]: updated the code in all the test cases as per the comment.

File: test/specs/mainnet/01_sendToken/Optimism.spec.js [hunk 3]
```diff
@@ -45,637 +50,555 @@ describe("The SDK, when sending a native token with optimism network on the Main
...
     }
+  });
 
-    // Estimating the batch
-    let estimationResponse;
-    let EstimatedGas_Estimate;
-    let FeeAmount_Estimate;
-    let EstimatedGasPrice_Estimate;
+  it("SMOKE: Perform the send native token on the optimism network", async () => {
+    if (runTest) {
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet/01_sendToken/Optimism.spec.js:89]: Noted!

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**